esp/services/ws_access/ws_accessService.cpp

```diff
@@ -1600,45 +1634,22 @@ bool Cws_accessEx::onPermissions(IEspContext &context, IEspBasednsRequest &req,
     return true;
 }
 
-const char* Cws_accessEx::getBaseDN(IEspContext &context, const char* rtype, StringBuffer& baseDN)
-{
-    if(!m_basedns.length())
-        setBasedns(context);
-    ForEachItemIn(y, m_basedns)
-    {
-        IEspDnStruct* curbasedn = &(m_basedns.item(y));
-        if(strieq(curbasedn->getRtype(), rtype))
-        {
-            baseDN.set(curbasedn->getBasedn());
-            return baseDN.str();
-        }
-    }
-    return NULL;
-}
-
 bool Cws_accessEx::onResources(IEspContext &context, IEspResourcesRequest &req, IEspResourcesResponse &resp)
 {
     try
     {
-        checkUser(context, req.getRtype(), req.getRtitle(), SecAccess_Read);
+        Owned<IEspDnStruct> basednReq = createDnStruct();
+        getBasednReq(context, req.getBasednName(), req.getBasedn(), req.getRtype(), req.getRtitle(), basednReq);
+
+        checkUser(context, basednReq->getRtype(), basednReq->getRtitle(), SecAccess_Read);
```

Comment:
why do you need to create a basednReq struct?  Can't you just get the rType and Rtitle from req?

Reply:
I prefer to use the structure which has three members. Right now, the getBasednReq() already has 6 input parameters. If the structure is not used, the getBasednReq() will have 8 input parameters.